sdk/compute/Microsoft.Azure.Management.Compute/src/Customizations/Disk.cs

```diff
@@ -0,0 +1,148 @@
+        /// will be allowed across all VMs mounting the shared disk as
+        /// ReadOnly. MBps means millions of bytes per second - MB here uses
+        /// the ISO notation, of powers of 10.</param>
+        /// <param name="diskState">The state of the disk. Possible values
+        /// include: 'Unattached', 'Attached', 'Reserved', 'Frozen',
+        /// 'ActiveSAS', 'ActiveSASFrozen', 'ReadyToUpload',
+        /// 'ActiveUpload'</param>
+        /// <param name="encryption">Encryption property can be used to encrypt
+        /// data at rest with customer managed keys or platform managed
+        /// keys.</param>
+        /// <param name="maxShares">The maximum number of VMs that can attach
+        /// to the disk at the same time. Value greater than one indicates a
+        /// disk that can be mounted on multiple VMs at the same time.</param>
+        /// <param name="shareInfo">Details of the list of all VMs that have
+        /// the disk attached. maxShares should be set to a value greater than
+        /// one for disks to allow attaching them to multiple VMs.</param>
+        /// <param name="networkAccessPolicy">Possible values include:
+        /// 'AllowAll', 'AllowPrivate', 'DenyAll'</param>
+        /// <param name="diskAccessId">ARM id of the DiskAccess resource for
+        /// using private endpoints on disks.</param>
+        /// <param name="tier">Performance tier of the disk (e.g, P4, S10) as
+        /// described here:
+        /// https://azure.microsoft.com/en-us/pricing/details/managed-disks/.
+        /// Does not apply to Ultra disks.</param>
+        /// <param name="burstingEnabled">Set to true to enable bursting beyond
+        /// the provisioned performance target of the disk. Bursting is
+        /// disabled by default. Does not apply to Ultra disks.</param>
+        /// <param name="propertyUpdatesInProgress">Properties of the disk for
+        /// which update is pending.</param>
+        /// <param name="supportsHibernation">Indicates the OS on a disk
+        /// supports hibernation.</param>
+        /// <param name="securityProfile">Contains the security related
+        /// information for the resource.</param>
+        /// <param name="supportedCapabilities">List of supported capabilities
+        /// for the image from which the OS disk was created.</param>
+        /// <param name="completionPercent">Percentage complete for the
+        /// background copy when a resource is created via the CopyStart
+        /// operation.</param>
+        /// <param name="publicNetworkAccess">Possible values include:
+        /// 'Enabled', 'Disabled'</param>
+        public Disk(string location, CreationData creationData, string id, string name, string type, IDictionary<string, string> tags, string managedBy, IList<string> managedByExtended, DiskSku sku, IList<string> zones, ExtendedLocation extendedLocation, System.DateTime? timeCreated, OperatingSystemTypes? osType, 
```

Comment:
the order of the parameters list should reflect the order of the parameters list in the default constructor before the new parameters were added